### ISSUE TITLE
K8s: Pod Container Status Metrics

### ISSUE THREAD

**Issue by daviddyball** [signalfx/signalfx-agent#944]
Hi,

`kube-state-metrics` includes two useful metrics for checking if you have pods stuck in `CrashLoopBackoff`, `ErrImagePull` or `ContainerCreating`:
* `kube_pod_container_status_running`
* `kube_pod_container_status_waiting_reason`
* `kube_pod_container_status_last_terminated_reason`

The docs for these metrics can be found [here](https://github.com/kubernetes/kube-state-metrics/blob/master/docs/pod-metrics.md)

It'd be really useful to not have to deploy `kube-state-metrics` in order to retrieve these metrics. Perhaps there is a way they could be put into the agent?

Cheers,
DD

**Comment by keitwb** [comment]
Hi David,

For `kube_pod_container_status_running`, you should be able to get that information from `kubernetes.pod_phase`.  The "running" state is the value `2`.

For the other two (wait/terminated reason), those are more appropriate as properties in our system, not so much metrics.  We will consider putting them as properties on the `kubernetes_pod_uid` dimension.

**Comment by daviddyball** [comment]
Good point on the `kubernetes.pod_phase`. We were already using it somewhere, I just overlooked it in this instance.

Are you sure `terminated` and `waiting` belong as dimensions on a metric? I'd think a simple gauge the same as `kubernetes.pod_phase` would be best suited to this, where each value means a different state.

**Comment by keitwb** [comment]
I would have to look more closely at what the possible set of values are for terminated/waiting.  If they are a fairly small well-established set of values that don't change across K8s releases much, they might be appropriate as an enum-like metric.  Do you know?

**Comment by daviddyball** [comment]
The spec in `kube-state-metrics` lists these as the known values:
```
OOMKilled
Error
Completed
ContainerCannotRun
DeadlineExceeded
```
It's not a long list, which is why an enum seemed like a good fit

**Comment by asuresh4** [comment]
Hi @daviddyball we've added two properties, `container_status` and `container_status_reason` onto `container_id` dimension from our `kubernetes-cluster` monitor. `container_status_reason` will be synced if `container_status` is either `terminated` or `waiting`. 

This update was released with v4.14.0. I'll keep this issue open for a couple of days - let us know if you have questions.

**Comment by daviddyball** [comment]
This is working as expected. Thanks @asuresh4
